tests/langchain/test_langchain_model_export.py

```diff
@@ -380,6 +383,104 @@ def load_retriever(persist_directory):
     )
 
 
+# Define a special embedding for testing
+class DeterministicDummyEmbeddings(Embeddings, BaseModel):
+    size: int
+
+    def _get_embedding(self, text: str) -> List[float]:
+        seed = abs(hash(text)) % (10**8)
+        np.random.seed(seed)
+        return list(np.random.normal(size=self.size))
+
+    def embed_documents(self, texts: List[str]) -> List[List[float]]:
+        return [self._get_embedding(t) for t in texts]
+
+    def embed_query(self, text: str) -> List[float]:
+        return self._get_embedding(text)
+
+
+def test_log_and_load_retriever_chain(tmp_path):
+    # Create the vector db, persist the db to a local fs folder
+    loader = TextLoader("tests/langchain/state_of_the_union.txt")
+    documents = loader.load()
+    text_splitter = CharacterTextSplitter(chunk_size=10, chunk_overlap=0)
+    docs = text_splitter.split_documents(documents)
+    embeddings = DeterministicDummyEmbeddings(size=5)
+    db = FAISS.from_documents(docs, embeddings)
+    persist_dir = str(tmp_path / "faiss_index")
+    db.save_local(persist_dir)
+
+    # Define the loader_fn
+    def load_retriever(persist_directory):
+        from typing import List  # pylint: disable=lazy-builtin-import
+
+        import numpy as np
+        from langchain.embeddings.base import Embeddings
+        from pydantic import BaseModel
+
+        class DeterministicDummyEmbeddings(Embeddings, BaseModel):
+            size: int
+
+            def _get_embedding(self, text: str) -> List[float]:
+                if isinstance(text, np.ndarray):
+                    text = text.item()
+                seed = abs(hash(text)) % (10**8)
+                np.random.seed(seed)
+                return list(np.random.normal(size=self.size))
+
+            def embed_documents(self, texts: List[str]) -> List[List[float]]:
+                return [self._get_embedding(t) for t in texts]
+
+            def embed_query(self, text: str) -> List[float]:
+                return self._get_embedding(text)
+
+        embeddings = DeterministicDummyEmbeddings(size=5)
+        vectorstore = FAISS.load_local(persist_directory, embeddings)
+        return vectorstore.as_retriever()
+
+    # Log the retriever
+    with mlflow.start_run():
+        logged_model = mlflow.langchain.log_model(
+            db.as_retriever(),
+            "retriever",
+            loader_fn=load_retriever,
+            persist_dir=persist_dir,
+        )
+
+    # Remove the persist_dir
+    shutil.rmtree(persist_dir)
+
+    # Load the RetrieverChain
+    loaded_model = mlflow.langchain.load_model(logged_model.model_uri)
+    assert loaded_model == RetrieverChain(retriever=db.as_retriever())
```

Comment:
The output of load model should be the retriever instance, not an instance of `RetrieverChain`. Can we update `load_model()` to unwrap if the loaded object is an instance of `RetrieverChain`?